src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -856,11 +878,10 @@ private void UpdateCacheExpirationTime(KeyValueWatcher changeWatcher)
             changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(cacheExpirationTime);
         }
 
-        private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IEnumerable<ConfigurationClient> clients, Func<ConfigurationClient, Task<T>> funcToExecute, CancellationToken cancellationToken = default)
+        private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IAsyncEnumerator<ConfigurationClient> clientEnumerator, Func<ConfigurationClient, Task<T>> funcToExecute, CancellationToken cancellationToken = default)
```

Comment:
```suggestion
        private async Task<T> ExecuteWithFailOverPolicyAsync<T>(IAsyncEnumerator<ConfigurationClient> clients, Func<ConfigurationClient, Task<T>> funcToExecute, CancellationToken cancellationToken = default)
```

Doesn't matter that the type changed. The intent is the same.